src/datalinker/dependencies/hips.py

```diff
@@ -56,39 +172,91 @@ async def _build_hips_list(
         logger
             Logger for any error messages.
         """
-        lists = []
-        for dataset in HIPS_DATASETS:
-            url = str(config.hips_base_url).rstrip("/") + f"/{dataset}"
-            r = await client.get(
-                url + "/properties",
-                headers={"Authorization": f"bearer {config.token}"},
+        config = config_dependency.config()
+
+        # Determine base URL: use default dataset if configured,
+        # otherwise legacy URL
+        if (
+            config.hips_default_dataset
+            and config.hips_default_dataset in config.hips_datasets
+        ):
+            dataset_config = config.hips_datasets[config.hips_default_dataset]
+            base_url = dataset_config.url
+            hips_paths = dataset_config.paths
+            logger.debug(
+                "Using default dataset for legacy endpoint",
+                dataset=config.hips_default_dataset,
             )
-            if r.status_code != 200:
-                logger.warning(
-                    "Unable to get HiPS list",
-                    url=url,
-                    status=r.status_code,
-                    error=r.reason_phrase,
-                )
-                continue
-            data = r.text
-
-            # Our HiPS properties files don't contain the URL
-            # (hips_service_url), but this is mandatory in the entries in the
-            # HiPS list.  Add it before hips_status.
-            service_url = "{:25}= {}".format("hips_service_url", url)
-            data = re.sub(
-                "^hips_status",
-                f"{service_url}\nhips_status",
-                r.text,
-                flags=re.MULTILINE,
+        else:
+            base_url = config.hips_base_url
+            hips_paths = HIPS_DATASETS
+            logger.debug(
+                "Using legacy hips_base_url for backward compatibility"
```

Comment:
I think it should be safe to drop all this and just require people configure datalinker correctly via Phalanx, unless I'm missing something. I'm usually pretty aggressive about breaking Phalanx-config-facing backwards compatibility in our services because we can change Phalanx at the same time as we roll out the new version and it's pretty safe to do.